marine/marine_pool.py

```diff
@@ -44,9 +43,29 @@ def filter_and_parse(
             chunksize=chunk_size,
         )
 
+    def set_preferences(self, preferences: List[str]) -> int:
+        # Because each process have a marine instance, and there is no way to tell each process to run
+        # the _marine_instance.set_preferences, we need to recreate the pool with the new configuration
+        self.preferences = preferences
+        self.pool.close()
```

Comment:
That's disgusting.
What's wrong with `self.pool.apply(cls._set_preferences, preferences)`?

Reply:
You can't